src/components/NucleotideEntropyMultiChart.tsx

```diff
@@ -0,0 +1,294 @@
+          {/* AA vs. nucs */}
+          <div className='mb-2 flex'>
+            <PipeDividedOptionsButtons
+              options={[
+                { label: 'NUC', value: 'nuc' },
+                { label: 'AA', value: 'aa' },
+              ]}
+              selected={sequenceType}
+              onSelect={setSequenceType}
+            />
+          </div>
+          {/* Include deletions */}
+            <FormGroup>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    checked={deletions}
+                    onChange={handleDeletions}
+                    inputProps={{ 'aria-label': 'controlled' }}
+                  />
+                }
+                label='Include deletions'
+              />
+            </FormGroup>
+          {/* Genes */}
+          {
+            <div className=' w-72 flex mb-2'>
+              <div className='mr-2'>Gene:</div>
+              <Form.Control
+                as='select'
+                value={gene}
+                onChange={ev => setGene(ev.target.value)}
+                className='flex-grow'
+                size='sm'
+              >
+                {jsonRefData.genes.map(g => (
+                  <option value={g.name} key={g?.name}>
+                    {g.name}
+                  </option>
+                ))}
+              </Form.Control>
+            </div>
+          }
+        </div>
+    );
+
+
+
+    let plotArea;
+    if (variants![0].length == 0) {
+      plotArea = <p>Select one or more lineages to compare.</p>
+    } else if (!data) {
+      plotArea = <Loader />;
+    } else {
+        plotArea = (
+        <Plot
+            plotData={data.dayArray}
+            ticks={data.ticks}
+            variants={variants!}
+            sequenceType={sequenceType}
+        />
+        );
+    }
+    
+    return (
+        <>
+        {controls}
+        {plotArea}
+        </>
+      );
+}
+
+type PlotProps = {
+    plotData: TransformedTime;
+    ticks: string[];
+    variants: string[];
+    sequenceType: SequenceType;
+};
+
+const useData = (
+    selectors: LapisSelector[],
+    selectedGene: GeneOption[],
+    variants: any,
+    sequenceType: SequenceType,
+    deletions: boolean
+): any | undefined => {
+  //fetch the proportions per position in weekly segments
+  const weeklyVariantMutationProportionQuery = useQuery(
+    async signal => {
+      //calculate weeks
+      const dayArray: UnifiedDay[] = [selectors[0].dateRange?.getDateRange().dateFrom!, selectors[0].dateRange?.getDateRange().dateTo!];
+      const dayRange = globalDateCache.rangeFromDays(dayArray)!;
+      const weeks = globalDateCache.weeksFromRange({ min: dayRange.min.isoWeek, max: dayRange.max.isoWeek });
+      const weekDateRanges = new Array<DateRange>();
+      for (let i = 0; i < weeks.length; i++) {
+        let dateRange: DateRange = { dateFrom: weeks[i].firstDay, dateTo: weeks[i].firstDay };
+        weekDateRanges[i] = dateRange;
+      }
```

Comment:
Using `Array.map` is usually more idiomatic are easier to read:
```suggestion
      const weekDateRanges: DateRange[] = globalDateCache
        .weeksFromRange({
          min: dayRange.min.isoWeek,
          max: dayRange.max.isoWeek,
        })
        .map(week => ({
          dateFrom: week.firstDay,
          dateTo: week.firstDay,
        }));
```